src/Microsoft.AspNetCore.Razor.Language/ClassifiedSpanVisitor.cs

```diff
@@ -0,0 +1,336 @@
+
+                var comment = razorCommentSyntax.Comment;
+                if (comment.IsMissing)
+                {
+                    // We need to generate a classified span at this position. So insert a marker in its place.
+                    comment = (SyntaxToken)SyntaxFactory.Token(SyntaxKind.Marker, string.Empty).Green.CreateRed(razorCommentSyntax, razorCommentSyntax.StartCommentStar.EndPosition);
+                }
+                WriteSpan(comment, SpanKindInternal.Comment, AcceptedCharactersInternal.Any);
+
+                WriteSpan(razorCommentSyntax.EndCommentStar, SpanKindInternal.MetaCode, AcceptedCharactersInternal.None);
+                WriteSpan(razorCommentSyntax.EndCommentTransition, SpanKindInternal.Transition, AcceptedCharactersInternal.None);
+
+                return razorCommentSyntax;
+            });
+        }
+
+        public override SyntaxNode VisitCSharpCodeBlock(CSharpCodeBlockSyntax node)
+        {
+            if (node.Parent is CSharpStatementBodySyntax ||
+                node.Parent is CSharpExplicitExpressionBodySyntax ||
+                node.Parent is CSharpImplicitExpressionBodySyntax ||
+                node.Parent is RazorDirectiveBodySyntax ||
+                (_currentBlockKind == BlockKindInternal.Directive &&
+                node.Children.Count == 1 &&
+                node.Children[0] is CSharpStatementLiteralSyntax))
+            {
+                return base.VisitCSharpCodeBlock(node);
+            }
+
+            return WriteBlock(node, BlockKindInternal.Statement, base.VisitCSharpCodeBlock);
+        }
+
+        public override SyntaxNode VisitCSharpStatement(CSharpStatementSyntax node)
+        {
+            return WriteBlock(node, BlockKindInternal.Statement, base.VisitCSharpStatement);
+        }
+
+        public override SyntaxNode VisitCSharpExplicitExpression(CSharpExplicitExpressionSyntax node)
+        {
+            return WriteBlock(node, BlockKindInternal.Expression, base.VisitCSharpExplicitExpression);
+        }
+
+        public override SyntaxNode VisitCSharpImplicitExpression(CSharpImplicitExpressionSyntax node)
+        {
+            return WriteBlock(node, BlockKindInternal.Expression, base.VisitCSharpImplicitExpression);
+        }
+
+        public override SyntaxNode VisitRazorDirective(RazorDirectiveSyntax node)
+        {
+            return WriteBlock(node, BlockKindInternal.Directive, base.VisitRazorDirective);
+        }
+
+        public override SyntaxNode VisitCSharpTemplateBlock(CSharpTemplateBlockSyntax node)
+        {
+            return WriteBlock(node, BlockKindInternal.Template, base.VisitCSharpTemplateBlock);
+        }
+
+        public override SyntaxNode VisitMarkupBlock(MarkupBlockSyntax node)
+        {
+            return WriteBlock(node, BlockKindInternal.Markup, base.VisitMarkupBlock);
+        }
+
+        public override SyntaxNode VisitGenericBlock(GenericBlockSyntax node)
```

Comment:
I already regret having a `GenericBlock`. This is currently only use in attribute values. I'm hoping I can get rid of this when I clean that up.

Reply:
Yeah there shouldn't be anything *generic*. Everything should be specific.